internal/module/kernelmapper.go

```diff
@@ -20,71 +19,72 @@ const (
 	kernelVersionPatchIdx = 2
 )
 
-type NodeOSConfig struct {
-	KernelFullVersion  string `subst:"KERNEL_FULL_VERSION"`
-	KernelVersionMMP   string `subst:"KERNEL_XYZ"`
-	KernelVersionMajor string `subst:"KERNEL_X"`
-	KernelVersionMinor string `subst:"KERNEL_Y"`
-	KernelVersionPatch string `subst:"KERNEL_Z"`
+type nodeOSConfig struct {
+	kernelFullVersion  string `subst:"KERNEL_FULL_VERSION"`
+	kernelVersionMMP   string `subst:"KERNEL_XYZ"`
+	kernelVersionMajor string `subst:"KERNEL_X"`
+	kernelVersionMinor string `subst:"KERNEL_Y"`
+	kernelVersionPatch string `subst:"KERNEL_Z"`
 }
 
-//go:generate mockgen -source=kernelmapper.go -package=module -destination=mock_kernelmapper.go
+//go:generate mockgen -source=kernelmapper.go -package=module -destination=mock_kernelmapper.go KernelMapper, kernelMapperHelperAPI
 
 type KernelMapper interface {
-	FindMappingForKernel(modSpec *kmmv1beta1.ModuleSpec, kernelVersion string) (*kmmv1beta1.KernelMapping, error)
-	GetNodeOSConfig(node *v1.Node) *NodeOSConfig
-	GetNodeOSConfigFromKernelVersion(kernelVersion string) *NodeOSConfig
-	PrepareKernelMapping(mapping *kmmv1beta1.KernelMapping, osConfig *NodeOSConfig) (*kmmv1beta1.KernelMapping, error)
+	GetProcessedMappingForKernel(modSpec *kmmv1beta1.ModuleSpec, kernelVersion string) (*kmmv1beta1.KernelMapping, error)
 }
 
 type kernelMapper struct {
-	buildHelper build.Helper
-	signHelper  sign.Helper
+	helper kernelMapperHelperAPI
 }
 
 func NewKernelMapper(buildHelper build.Helper, signHelper sign.Helper) KernelMapper {
+	helper := newKernelMapperHelper(buildHelper, signHelper)
 	return &kernelMapper{
-		buildHelper: buildHelper,
-		signHelper:  signHelper,
+		helper: helper,
 	}
 }
 
-// FindMappingForKernel tries to match kernelVersion against mappings. It returns the first mapping that has a Literal
-// field equal to kernelVersion or a Regexp field that matches kernelVersion.
-func (k *kernelMapper) FindMappingForKernel(modSpec *kmmv1beta1.ModuleSpec, kernelVersion string) (*kmmv1beta1.KernelMapping, error) {
+func (k *kernelMapper) GetProcessedMappingForKernel(modSpec *kmmv1beta1.ModuleSpec, kernelVersion string) (*kmmv1beta1.KernelMapping, error) {
 	mappings := modSpec.ModuleLoader.Container.KernelMappings
-	foundMapping, err := k.findKernelMapping(mappings, kernelVersion)
+	foundMapping, err := k.helper.findKernelMapping(mappings, kernelVersion)
 	if err != nil {
 		return nil, fmt.Errorf("failed to find mapping for kernel %s: %v", kernelVersion, err)
 	}
 	mapping := foundMapping.DeepCopy()
-
-	// prepare the build for the mapping
-	if mapping.Build != nil || modSpec.ModuleLoader.Container.Build != nil {
-		mapping.Build = k.buildHelper.GetRelevantBuild(modSpec.ModuleLoader.Container.Build, mapping.Build)
+	err = k.helper.prepareMappingData(mapping, modSpec, kernelVersion)
```

Comment:
`prepare` is quite vague IMO. How about:
```suggestion
	err = k.helper.unifyMappingData(mapping, modSpec, kernelVersion)
```

Reply:
fixed